k8s/cassandra/apptest/tester/tests/basic-suite.yaml

```diff
@@ -0,0 +1,52 @@
+actions:
+- name: 3 nodes deployed successfully
+  bashTest:
+    script: |-
+      set -x
+      function info {
+        >&2 echo "${@}"
+      }
+
+      function get_desired_number_of_replicas_in_sts {
+        kubectl get sts "${STS_NAME}" \
+          --namespace "${NAMESPACE}" \
+          --output jsonpath='{.spec.replicas}'
+      }
+
+      function get_current_number_of_replicas_in_sts {
+        kubectl get sts "${STS_NAME}" \
+          --namespace "${NAMESPACE}" \
+          --output jsonpath='{.status.readyReplicas}'
+      }
+
+      function wait_for_healthy_sts {
+        info "Waiting for equal desired and current number of replicas"
+        while [[ $(get_current_number_of_replicas_in_sts) -ne $(get_desired_number_of_replicas_in_sts) ]]; do
```

Comment:
No action: ideally this should have a time out, but we can use the deadline of the encompassing Job.

Reply:
Well... I do not think that we should have a timeout on each layer. If we will have too many of them, then increasing a timeout will take a lots of manual work to recheck the code.

I've added a timeout on job level, I think that this is enough for our use case.